tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover0",
+                CONF_IP_ADDRESS: "127.0.1.0",
+                CONF_USERNAME: "user0",
+                CONF_PASSWORD: "password0",
+            },
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover1",
+                CONF_IP_ADDRESS: "127.0.1.1",
+                CONF_USERNAME: "user1",
+                CONF_PASSWORD: "password1",
+            },
+        ]
+    )
+    entity_ids = hass.states.async_entity_ids(COVER_DOMAIN)
+    assert entity_ids is not None
+    assert len(entity_ids) == 2
+    assert "cover.door1" in entity_ids
+    assert "cover.door1_2" in entity_ids
+
+    await component_factory.unload()
+
+
+async def test_cover_update(
+    hass: HomeAssistant, component_factory: ComponentFactory
+) -> None:
+    """Test cover."""
+    await component_factory.configure_component()
+    component_data = await component_factory.run_config_flow(
+        config_data={
+            CONF_IP_ADDRESS: "127.0.0.2",
+            CONF_USERNAME: "user0",
+            CONF_PASSWORD: "password0",
+        }
+    )
+
+    assert hass.states.async_entity_ids(COVER_DOMAIN)
+
+    state = hass.states.get("cover.door1")
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes["friendly_name"] == "Door1"
+    assert state.attributes["supported_features"] == 3
+    assert state.attributes["device_class"] == "garage"
+
+    component_data.data_update_coordinator.api.info.return_value = InfoResponse(
+        user="user1",
+        gogogatename="gogogatename0",
+        model="",
+        apiversion="",
+        remoteaccessenabled=False,
+        remoteaccess="abc123.blah.blah",
+        firmwareversion="",
+        apicode="",
+        door1=Door(
+            door_id=1,
+            permission=True,
+            name="Door1",
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.OPENED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=2,
+            temperature=None,
+        ),
+        door2=Door(
+            door_id=2,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+    await component_data.data_update_coordinator.async_refresh()
```

Comment:
Don't call the coordinator in the tests. That's a detail that may change in the future. Patch time and fire a time changed event to take control over time and move it forward as needed.

Reply:
See eg this test:
https://github.com/home-assistant/core/blob/65e509ed8f8791409072b9f1281cd70279367989/tests/components/gdacs/test_geo_location.py#L43